### PR TITLE
VOLDEV-186: Whitelist Gadget-* to make the gadget extension usable

### PR DESCRIPTION
**Pull request by onei**
Also whitelists Gadgets-*, which applies to messages defined by the extension itself, thus allowing users to alter gadgets-definition as well.

@Grunny

### CODE REVIEW

**Comment by mszabo-wikia** [comment]
CR: https://wikia-inc.atlassian.net/browse/SUS-1548

File: extensions/Gadgets/Gadgets_body.php [hunk 1]
```diff
@@ -87,7 +87,10 @@ public static function getPreferences( $user, &$preferences ) {
 			}
 
 			if ( $section !== '' ) {
-				$section = wfMsgExt( "gadget-section-$section", 'parseinline' );
+				// begin wikia change
+				// wfMsg cleanup
+				$section = wfMessage( "gadget-section-$section" )->parse();
```
Comment (mszabo-wikia) [inline extensions/Gadgets/Gadgets_body.php]: We should consider either adding separate use for each possible value of `$section`, or at least list the possible values in comments - helps when searching for usage of messages

File: extensions/Gadgets/SpecialGadgets.php [hunk 2]
```diff
@@ -188,7 +208,10 @@ public function showExportForm( $gadget ) {
 			. Html::hidden( 'pages', $exportList )
 			. Html::hidden( 'wpDownload', '1' )
 			. Html::hidden( 'templates', '1' )
-			. Xml::submitButton( wfMsg( 'gadgets-export-download' ) )
+			// begin wikia change
+			// wfMsg cleanup
+			. Xml::submitButton( wfMessage( 'gadgets-export-download' )->escaped() )
```
Comment (mszabo-wikia) [inline extensions/Gadgets/SpecialGadgets.php]: This function escapes its parameter, so no need for double escaping here

Also this is inside a special page so we can use `$this->msg` to set proper request context

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]

File: extensions/Gadgets/Gadgets_body.php (hunk 1 above)
Comment (onei) [reply extensions/Gadgets/Gadgets_body.php]: In this case, the name of the message is user defined allowing them to group similar gadgets under appropriate heads on gadgets-definition and Special:Gadgets. If you wanted a list of them, you'd have to look them up on Special:PrefixIndex on the wiki the extension is installed on or something similar.

For reference, the messages defined by the extension are prefixed with gadgets-*.

File: extensions/Gadgets/SpecialGadgets.php (hunk 2 above)
Comment (onei) [reply extensions/Gadgets/SpecialGadgets.php]: Done in 21954ee

**Review by macbre — APPROVED** [review]

**Comment by macbre** [comment]
Ok, let's merge this one into a branch in our upstream repository. Once it passes QA we can merge it into our dev.

**Pull request merged**